### PR TITLE
124 feature multilingual website

### LINKED ISSUE #124: Feature: Multilingual Website
## Summary

Let's make the translate button active. Translate all the static text in the website and put them in their own files in public/assets/locales. Use the useTranslate hook from i18n to make the website multilingual and the language button active.

### Screenshots

N/A

### Basic example

N/A

### Motivation

To have a multilingual website.

### PR DESCRIPTION
**Pull request by ayubabdullah**
adding functionality to change language button and prepare three translation files, Arabic and Kurdish translation is not done yet, I'm working on it, if anyone want to help me in these files please text me I appreciated

### CODE REVIEW

**Review by OussamaMasri — APPROVED** [review]
Great Job Ayub !
Very well done

File: public/assets/locales/en/translation.json [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        "item_type": "Item Type",
+        "item_category": "Item Category",
+        "item_description": "Description",
+        "adress_info": "Address Info",
+        "country": "Country",
+        "city": "City",
+        "add_new_item": "Add New Item"
+    },
+    "overlap_text": {
+        "check_out": "Watch Full Video",
```
Comment (sarahaboudalal) [inline public/assets/locales/en/translation.json]: this one was "check out this video made by one of our satisfied customers" . i suggest we don't change it

Comment (sarahaboudalal) [inline public/assets/locales/en/translation.json]: this one was "check out this video made by one of our satisfied customers" . i suggest we don't change it

File: public/assets/locales/ar/translation.json [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        "feedback_two_title": "عميل",
+        "feedback_two_description": "شعرت كل عملية شراء بالأمان وتم استلام جميع العناصر في الوقت المناسب. يا له من مكان رائع لعرض منتجات وسلع العمل!",
+        "feedback_two_name": "Steve Mark",
+        "feedback_three_title": "عميل",
+        "feedback_three_description": "لقد وجدت أن دلمون كانت محترفة للغاية وقدمت منتجًا عالي الجودة في إطار زمني فعال. سأفكر في التسوق معهم مرة أخرى.",
+        "feedback_three_name": "Kenn Gallagher"
+    },
+    "search_bar": {
+        "search_here": "ابحث هنا...",
+        "submit": "يُقدِّم"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: let this be "ارسال

File: public/assets/locales/ar/translation.json [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    "news_letter": {
+        "subscribe_to_our_newsletter": "اشترك في النشرة<br /> الإخبارية",
+        "description": "اشترك في النشرة الإخبارية لدينا وابقى على اطلاع",
+        "subscribe": "اشترك"
+    },
+    "proof": {
+        "our_partners": "شركائنا",
+        "what_they_say": "ما يقوله الناس <br className='block' /> عنا",
+        "everything_you_need": " كل ما تحتاج لمعرفته حول أعمالنا ومنتجاتنا في أي مكان على هذا الكوكب.",
+        "feedback_one_title": "عميل",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: زبون

File: public/assets/locales/ar/translation.json [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        "apply": "يتقدم"
+    },
+    "hero": {
+        "welcome_to_dilmun": "مرحبا بكم في دلمون",
+        "your_way": "طريقك للحصول على منتجات عالية الجودة!",
+        "get_started": "البدء"
+    },
+    "edit_item_form": {
+        "edit_item": "تعديل عنصر",
+        "edit_item_info": "تحرير معلومات العنصر",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json:90]: instead of تحرير let it be  تعديل

File: public/assets/locales/ar/translation.json [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        },
+        "type": {
+            "title": "أنواع",
+            "new": "الجديد",
+            "used": "تستخدم",
+            "donated": "تبرعت",
+            "crafted": "مصنوع"
+        },
+        "price": "سعر",
+        "apply": "يتقدم"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: طبّق

File: public/assets/locales/ar/translation.json [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+            "women": "النساء",
+            "kids": "أطفال",
+            "toys": "ألعاب الأطفال",
+            "electronics": "إلكترونيات",
+            "home": "عنصر المنزل"
+        },
+        "type": {
+            "title": "أنواع",
+            "new": "الجديد",
+            "used": "تستخدم",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: مستخدم

File: public/assets/locales/ar/translation.json [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            "donated": "تبرعت",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: للتبرع

File: public/assets/locales/ar/translation.json [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+            "crafted": "مصنوع"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: صناعة يدوية

File: public/assets/locales/ar/translation.json [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+            "title": "الفئة",
+            "men": "رجال",
+            "women": "النساء",
+            "kids": "أطفال",
+            "toys": "ألعاب الأطفال",
+            "electronics": "إلكترونيات",
+            "home": "عنصر المنزل"
+        },
+        "type": {
+            "title": "أنواع",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json:64]: عنوان

File: public/assets/locales/ar/translation.json [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+        "question_five_answer": "سيكون فريق خدمة العملاء لدينا متواجدًا دائمًا لحل مشكلتك ، يمكنك الاتصال بنا."
+    },
+    "features": {
+        "designed_to_meet_your_needs": "مصممة لتلبية احتياجاتك.",
+        "description": "هنا في دلمون نبيع العديد من المنتجات للعملاء ونقبل التبرعات لمساعدة المحتاجين.",
+        "feature_one_title": "منقي",
+        "feature_one_description": "تصفية العناصر الخاصة بك للعثور على المعلومات الصحيحة.",
+        "feature_two_title": "متعدد اللغات",
+        "feature_two_description": "استخدم الموقع بلغتك المفضلة.",
+        "feature_three_title": "يجتمع",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: قابل\اجتمع

Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: قابل\اجتمع

File: public/assets/locales/ar/translation.json [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+        "question_three_answer": "يمكنك الدفع عبر الإنترنت عن طريق بطاقة الائتمان أو الدفع عند الباب أثناء التعامل مع البائع.",
+        "question_four_title": "كيف أتتبع طلبي؟",
+        "question_four_answer": "يمكنك الحصول على رقم هاتف الشركة التي تقوم بالتوصيل واسألهم عن مكان طلبك.",
+        "question_five_title": "ما هي سياسة التأخر في تسليم / عدم تسليم العناصر المطلوبة عبر الإنترنت؟",
+        "question_five_answer": "سيكون فريق خدمة العملاء لدينا متواجدًا دائمًا لحل مشكلتك ، يمكنك الاتصال بنا."
+    },
+    "features": {
+        "designed_to_meet_your_needs": "مصممة لتلبية احتياجاتك.",
+        "description": "هنا في دلمون نبيع العديد من المنتجات للعملاء ونقبل التبرعات لمساعدة المحتاجين.",
+        "feature_one_title": "منقي",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: فلتر idk if that's even arabic but everyone use that lol

File: public/assets/locales/ar/translation.json [hunk 12]
```diff
@@ -0,0 +1,168 @@
+{
+    "about_us": {
+        "meet_our_team": "التق بفريقنا"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: تعرف على فريقنا

File: public/assets/locales/ar/translation.json [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+    "add_item_form": {
+        "product_info": "معلومات المنتج",
+        "note": "ستتم إضافة منتجك إلى المتجر بعد تعبئة البيانات المطلوبة.",
+        "item_image": "صورة البند",
+        "upload_a_file": "قم بتحميل ملف",
+        "or_drag_and_drop": "أو السحب والإفلات",
+        "item_name": "اسم العنصر",
+        "item_price": "سعر",
+        "item_type": "نوع العنصر",
+        "item_category": "العنصر فئة",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: فئة العنصر

File: public/assets/locales/ar/translation.json [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+    "filter": {
+        "filters": "المرشحات",
+        "category": {
+            "title": "الفئة",
+            "men": "رجال",
+            "women": "النساء",
+            "kids": "أطفال",
+            "toys": "ألعاب الأطفال",
+            "electronics": "إلكترونيات",
+            "home": "عنصر المنزل"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: ادوات منزلية they also should be households in english

File: public/assets/locales/ar/translation.json [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+            "donated": "تبرعت",
+            "crafted": "مصنوع"
+        },
+        "price": "سعر",
+        "apply": "يتقدم"
+    },
+    "hero": {
+        "welcome_to_dilmun": "مرحبا بكم في دلمون",
+        "your_way": "طريقك للحصول على منتجات عالية الجودة!",
+        "get_started": "البدء"
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json]: ابدأ الآن

File: public/assets/locales/ar/translation.json [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+        "best_rating": "افضل تقييم",
+        "trending": "الشائع",
+        "last_7_days": "اخر 7 ايام",
+        "last_30_days": "آخر 30 يومًا",
+        "coming_soon": "قريبا",
+        "price_free": "مجاني",
+        "price_low_to_high": "السعر من الارخص للاعلى",
+        "price_high_to_low": "السعر الاعلى الى الادنى"
+    },
+    "not_found": "ربما انتقلت هذه الصفحة؟ هل تم حذفه؟ <br /> هل يختبئ في الحجر الصحي؟ لم تكن موجودة في المقام الأول؟",
```
Comment (sarahaboudalal) [inline public/assets/locales/ar/translation.json:180]: تختبئ and  حذفها

**Review by sarahaboudalal — CHANGES_REQUESTED** [review]
don't even ask me how i did that but i just did it lol. check my comments. but WELL DONE it's epic @ayubabdullah

**Comment by ayubabdullah** [comment]
> don't even ask me how i did that but i just did it lol. check my comments. but WELL DONE it's epic @ayubabdullah

For the Arabics i used Google translate so any mistake blame google not me LOL 😂 I'll fix them

**Comment by Iseluin** [comment]
Since this will be a constant WIP until we finish the website, I'm merging this at the end. Sounds good @ayubabdullah ?

**Comment by ayubabdullah** [comment]
> Since this will be a constant WIP until we finish the website, I'm merging this at the end. Sounds good @ayubabdullah ?

Yep boss sounds good, i will keep updating it until all the components are finished

**Review by sarahaboudalal — APPROVED** [review]

**Pull request merged**